src/emulator/gui/src/settings_dialog.cpp

```diff
@@ -15,53 +15,61 @@
 // with this program; if not, write to the Free Software Foundation, Inc.,
 // 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 
-#include "private.h"
-#include <config.h>
-#include <gui/functions.h>
-
-#include <host/functions.h>
-#include <host/state.h>
-#include <misc/cpp/imgui_stdlib.h>
-#include <util/fs.h>
-#include <util/log.h>
+#include <gui/settings_dialog.h>
 
 namespace gui {
 
-bool change_pref_location(const std::string &input_path, const std::string &current_path) {
-    if (fs::path(input_path).has_extension())
-        return false;
-
-    if (!fs::exists(input_path))
-        fs::create_directories(input_path);
-
-    try {
-        fs::directory_iterator it{ current_path };
-        while (it != fs::directory_iterator{}) {
-            // TODO: Move Vita directories
-
-            boost::system::error_code err;
-            it.increment(err);
-        }
-    } catch (const std::exception &err) {
-        return false;
-    }
-    return true;
-}
-
 void draw_settings_dialog(HostState &host) {
     ImGui::PushStyleColor(ImGuiCol_Text, GUI_COLOR_TEXT_MENUBAR_OPTIONS);
     ImGui::Begin("Settings", &host.gui.configuration_menu.settings_dialog, ImGuiWindowFlags_AlwaysAutoResize);
     ImGuiTabBarFlags settings_tab_flags = ImGuiTabBarFlags_None;
     ImGui::BeginTabBar("SettingsTabBar", settings_tab_flags);
 
+    // Core
+    if (ImGui::BeginTabItem("Core")) {
+        ImGui::PopStyleColor();
+        if (vfs::read_file(VitaIoDevice::VS0, buffer, host.pref_path, "sys/external/libc.suprx")) {
+            ImGui::Text("Modules List\nSelect your desired modules.");
+            ImGui::PushItemWidth(240);
+            if (ImGui::ListBoxHeader("##modules_list", IM_ARRAYSIZE(LIST_MODULES), 8)) {
```

Comment:
```suggestion
            constexpr int MODULES_COUNT = IM_ARRAYSIZE(LIST_MODULES);
            if (ImGui::ListBoxHeader("##modules_list", MODULES_COUNT, 8)) {
```

Since you are reusing that phrase

Reply:
Why remove listboxheader ? I have set it for get it works like recommanded imgui creator for using with selectable

Reply:
I dont tell you to remove it. Look carefully.

Reply:
ah yes wrong see lol sorry